Tests/SourceKitLSPTests/CodeActionTests.swift

```diff
@@ -299,6 +299,48 @@ final class CodeActionTests: XCTestCase {
     )
   }
 
+  func testJSONCodableCodeActionResult() async throws {
+    let testClient = try await TestSourceKitLSPClient(capabilities: clientCapabilitiesWithCodeActionSupport())
+    let uri = DocumentURI.for(.swift)
+    let positions = testClient.openDocument(
+      """
+         1️⃣{
+         "name": "Produce",
+         "shelves": [
```

Comment:
I think right now it’s only possible to invoke the refactoring from the opening `{`. I think it would be a lot more discoverable if you could invoke it from anywhere within the JSON.

Reply:
No, the refactoring can be invoked from elsewhere in the closure. The test harness doesn't make it easy to check for places other than the start, though.

Reply:
Hmm, I see what's going on. The parser isn't treating the whole blob as a single closure, so we sometimes end up in the "unexpected" area. I don't have a great fix at the moment.

Reply:
Could you file an issue for that?